app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
Okay, make sense. I did it that way because I read some Rails tutorials which keep using `set_ABC` to set the instance variable `ABC` instead of just using `ABC`. So I thought it's a thing in Rails lor.